entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/EntityViewManagerImpl.java

```diff
@@ -244,9 +244,56 @@ public ProxyFactory getProxyFactory() {
     public <T> T find(EntityManager entityManager, EntityViewSetting<T, CriteriaBuilder<T>> entityViewSetting, Object entityId) {
         ViewTypeImpl<T> managedViewType = metamodel.view(entityViewSetting.getEntityViewClass());
         EntityType<?> entityType = (EntityType<?>) managedViewType.getJpaManagedType();
-        SingularAttribute<?, ?> idAttribute = JpaMetamodelUtils.getSingleIdAttribute(entityType);
-        CriteriaBuilder<?> cb = cbf.create(entityManager, managedViewType.getEntityClass())
-                .where(idAttribute.getName()).eq(entityId);
+        Set<SingularAttribute<?, ?>> idAttributeSet = JpaMetamodelUtils.getIdAttributes(entityType);
+        CriteriaBuilder<?> cb = cbf.create(entityManager, managedViewType.getEntityClass());
+        Map<String,Object> entityIdValues = new HashMap<>();
+        //TODO Hibernate returns null when getIdType it is called on an entityType which uses IdClass.
+        if(entityType.getIdType()!=null && (entityType.getIdType().getPersistenceType().equals(Type.PersistenceType.BASIC)||
+                entityType.getIdType().getPersistenceType().equals(Type.PersistenceType.EMBEDDABLE))){
+            Iterator iterator = idAttributeSet.iterator();
+            if(!iterator.hasNext()){
+                throw new RuntimeException("The entity type" + entityType.getName() + "does not have an Id!");
+                }
+
+            //I believe this approach also works for BigDecimal and the like? I have added that possibility in the check above.
+            entityIdValues.put(((SingularAttribute) iterator.next()).getName(),entityId);
+
+            if (iterator.hasNext()){
+                throw new RuntimeException("Could not match the given entityId to the entity type specified in the view: the entity type"
+                        + entityType.getName() + "has more than one Id field!");
+            }
+        } else {
+//            for (Field field : Arrays.asList(entityId.getClass().getDeclaredFields())){
+//                field.setAccessible(Boolean.TRUE);
+//                try {
+//                    entityIdValues.put(field.getName(),field.get(entityId));
+//                } catch (IllegalAccessException e) {
+//                    throw new IllegalStateException("Could not access field: "+field.getName()+".");
+//                }
+//            }
+//            entityIdValues.remove("serialVersionUID");
+            for (SingularAttribute<?,?> idAttribute : idAttributeSet){
+                Method method = (Method) idAttribute.getJavaMember();
```

Comment:
I think that the `Member` can also be of type `Field`, in which case you do not want to cast it to a `Method`, but rather cast it to a `Field`, and then do `field.get(entityId)` instead of `method.invoke(entityId)`. Actually, I am worried that theres even much more to it than that. See for example the code in `JpaMetamodelUtils.resolveFieldClass(basClass, attr)`, which looks up the type of the field of an attribute.